src/main/java/se/sundsvall/agreement/api/PagedAgreementResource.java

```diff
@@ -0,0 +1,55 @@
+package se.sundsvall.agreement.api;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.zalando.problem.Problem;
+import org.zalando.problem.violations.ConstraintViolationProblem;
+import se.sundsvall.agreement.api.model.AgreementParameters;
+import se.sundsvall.agreement.api.model.AgreementResponse;
+import se.sundsvall.agreement.api.model.Category;
+import se.sundsvall.agreement.api.model.PagedAgreementResponse;
+import se.sundsvall.agreement.service.AgreementService;
+import se.sundsvall.dept44.common.validators.annotation.ValidUuid;
+
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.MediaType.APPLICATION_PROBLEM_JSON_VALUE;
+
+@RestController
+@Validated
+@RequestMapping("paged/agreements")
+@Tag(name = "Paged Agreement", description = "Agreement resources with paging")
+public class PagedAgreementResource {
+
+	@Autowired
+	private AgreementService agreementService;
+
+	@GetMapping(path = "/{partyId}", produces = { APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE })
+	@Operation(summary = "Get agreements connected to a party-ID, optionally filtered by provided categories")
+	@ApiResponse(responseCode = "200", description = "Successful operation", content = @Content(mediaType = APPLICATION_JSON_VALUE, schema = @Schema(implementation = AgreementResponse.class)))
```

Comment:
Schema defined response does not match actual response. 
`useReturnTypeSchema = true` ?